list-doi-requests/src/main/java/no/unit/nva/doirequest/list/ListDoiRequestsHandler.java

```diff
@@ -85,22 +111,67 @@ private boolean userHasRequestedRole(String role, String userRolesCsv) {
     }
 
     private Publication[] fetchDoiRequestsForCurator(UserInstance userInstance) {
         List<DoiRequest> doiRequests = doiRequestService.listDoiRequestsForPublishedPublications(userInstance);
+        List<Publication> publicationDtos = convertInternalObjectsToDtos(doiRequests);
+        List<Publication> enrichedWithMessages = enrichPublicationDtosWithDoiRequestMessages(publicationDtos);
+        return convertListToArray(enrichedWithMessages);
+    }
+
+    private List<Publication> enrichPublicationDtosWithDoiRequestMessages(List<Publication> dtos) {
+        return dtos.stream()
+                   .map(this::addMessagesToPublicationDto)
+                   .sorted(sortByOldestDoiRequestFirst())
+                   .collect(Collectors.toList());
+    }
+
+    private Comparator<Publication> sortByOldestDoiRequestFirst() {
+        return Comparator.comparing(p -> p.getDoiRequest().getCreatedDate());
+    }
+
+    private Publication addMessagesToPublicationDto(Publication dto) {
+        Stream<ResourceMessages> messages =
+            messageService.getMessagesForResource(extractOwner(dto), dto.getIdentifier()).stream();
+        List<DoiRequestMessage> doiRequestMessages = transformToLegacyDoiRequestMessagesDto(messages);
+        dto.getDoiRequest().setMessages(doiRequestMessages);
+        return dto;
+    }
+
+    private List<DoiRequestMessage> transformToLegacyDoiRequestMessagesDto(Stream<ResourceMessages> messages) {
+        return messages
+                   .flatMap(resourceMessages -> resourceMessages.getMessages().stream())
+                   .filter(MessageDto::isDoiRequestRelated)
+                   .map(this::toDoiRequestMessage)
+                   .sorted(sortByTimeOldestFirst())
+                   .collect(Collectors.toList());
+    }
 
-        return convertInternalObjectsToDtos(doiRequests);
+    private Comparator<DoiRequestMessage> sortByTimeOldestFirst() {
+        return Comparator.comparing(DoiRequestMessage::getTimestamp);
     }
 
-    private Publication[] convertInternalObjectsToDtos(List<DoiRequest> doiRequests) {
-        List<Publication> dtos = doiRequests.stream()
-            .map(DoiRequest::toPublication)
-            .collect(Collectors.toList());
-        Publication[] result = new Publication[doiRequests.size()];
+    private DoiRequestMessage toDoiRequestMessage(MessageDto message) {
+        return new DoiRequestMessage.Builder()
+                   .withText(message.getText())
+                   .withTimestamp(message.getDate())
+                   .withAuthor(message.getSenderIdentifier())
+                   .build();
+    }
+
+    private Publication[] convertListToArray(List<Publication> dtos) {
+        Publication[] array = new Publication[dtos.size()];
+        dtos.toArray(array);
+        return array;
+    }
 
-        dtos.toArray(result);
-        return result;
+    private List<Publication> convertInternalObjectsToDtos(List<DoiRequest> doiRequests) {
```

Comment:
what is an "internal object"?

Reply:
For historical reasons, the handler returns a Publication (DTO)  and not a DoiRequest. This is wrong as the query is asking for DoiRequests and messages. The reason why this is happening is because of a hotfix that we did and we never addressed it after that in the light of this refactoring. In this phase of the refactoring I want to maintain the contract as much as possible and therefore I return a Publication. 

Now, the DTO is not produced by one DAO but by an intermediate Object that collects informaiton from multiple DAOs. I have no good name for that object.